src/core/lib/transport/handshaker_factory.h

```diff
@@ -35,11 +37,38 @@ namespace grpc_core {
 
 class HandshakeManager;
 
+// Enum representing the priority of the client handshakers.
+// The order of the client handshakers is decided by the priority.
+// For example kPreTCPConnect handshakers are called before kTCPConnect and so
+// on.
+enum class HandshakerClientPriority : int {
+  kPreTCPConnect,
+  kTCPConnect,
+  kHTTPConnect,
+  kSecurity,
+};
+
+// Enum representing the priority of the server handshakers.
+// The order of the server handshakers is decided by the priority.
+// For example kReadHeadSecurity handshakers are called before kSecurity and so
+// on.
+enum class HandshakerServerPriority : int {
+  kReadHeadSecurity,
+  kSecurity,
+};
+
+using HandshakerPriority =
```

Comment:
This is probably reaching for a too large a hammer for the job.

Consider folding HandshakerClientPriority/HandshakerServerPriority into one enum ... we can comment client/server only if necessary (heck, we can introduce a IsServer/IsClient if necessary too... but I don't think it's practically useful)

Reply:
I initially had a single priority for both clients and servers but I thought it looked odd to have PreTCPConnectHandshaker for Server side handshakers. Adding a comment sounds good to me considering the current limited usage.